go/websocketbridge/websocket_bridge_test.go

```diff
@@ -0,0 +1,202 @@
+package websocketbridge_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"log"
+	"net/http"
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/DATA-DOG/godog"
+	"github.com/DATA-DOG/godog/gherkin"
+	"github.com/Originate/exocom/go/exocom-mock"
+	"github.com/Originate/exocom/go/structs"
+	"github.com/Originate/exocom/go/websocketbridge"
+
+	"golang.org/x/net/websocket"
+)
+
+func newExocom(port int) *exocomMock.ExoComMock {
+	exocom := exocomMock.New()
+	go func() {
+		err := exocom.Listen(port)
+		if err != nil && err != http.ErrServerClosed {
+			log.Fatal(err)
+		}
+	}()
+	return exocom
+}
+
+// Cucumber step definitions
+// nolint gocyclo
+func FeatureContext(s *godog.Suite) {
+	var exocom *exocomMock.ExoComMock
+	var websocketBridgeInstance *websocketbridge.WebsocketBridge
+	var ws *websocket.Conn
+	exoPort := 4100
+	clientPort := 5435
+	var outgoingMessageId string
+	var savedError error
+
+	s.BeforeSuite(func() {
+		exocom = newExocom(exoPort)
+	})
+
+	s.BeforeScenario(func(interface{}) {
+		outgoingMessageId = ""
+		savedError = nil
+		ws = nil
+	})
+
+	s.AfterScenario(func(interface{}, error) {
+		exocom.Reset()
+		err := websocketBridgeInstance.Reset()
+		if err != nil {
+			panic(err)
+		}
+	})
+
+	s.AfterSuite(func() {
+		err := exocom.Close()
+		if err != nil {
+			panic(err)
+		}
+	})
+
+	s.Step(`a websocket bridge connected to Exocom`, func() error {
+		websocketBridgeInstance = &websocketbridge.WebsocketBridge{
+			ClientPort: clientPort,
+			Config: websocketbridge.Config{
+				Host: "localhost",
+				Port: exoPort,
+				Role: "websocket-test",
+			},
+		}
+		err := websocketBridgeInstance.ConnectToExocom()
+		if err != nil {
+			return err
+		}
+		err = websocketBridgeInstance.ListenForClients()
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+
+	s.Step(`receiving this message from the client:`, func(payloadStr *gherkin.DocString) error {
+		origin := "http://localhost/"
+		url := fmt.Sprintf("ws://localhost:%v/ws", clientPort)
+		var err error
+		ws, err = websocket.Dial(url, "", origin)
+		if err != nil {
+			log.Fatal(err)
```

Comment:
Please return the err instead of using `log.Fatal`. Step definitions can return an error which godog would report instead of this killing the suite